music21/figuredBass/notation.py

```diff
@@ -410,15 +518,34 @@ class Figure(prebase.ProtoM21Object):
             associated with an expanded
             :attr:`~music21.figuredBass.notation.Notation.notationColumn`.
             ''',
+        'hasExtender': '''
+            A bool value that indicates whether an extender is part of the figure.
+            It is set by a keyword argument.
+            ''',
+        'isExtender': '''
+            A bool value that returns true if an extender is part of the figure but no
+            number is given. Pure extender if you will.
+            It is set by evaluating the number and extender arguments.
+            '''
     }
 
-    def __init__(self, number=1, modifierString=None):
+    isExtender: bool
+
+    def __init__(self, number=1, modifierString=None, extender=False):
         self.number = number
         self.modifierString = modifierString
         self.modifier = Modifier(modifierString)
+        # look for extender's underscore
+        self.hasExtender: bool = extender
+        self._updateIsExtenderProperty()
+        
+    def _updateIsExtenderProperty(self):
+        self.isExtender = (self.number == 1 and self.hasExtender)
```

Comment:
what I meant is if I create a figure object like:

```
>>> n = notation.Figure(1, '#', extender=True)
```

then the `.isExtender` property can get out of date:

```
>>> n.isExtender
True
>>> n.hasExtender = False
>>> n.isExtender
True
```

instead don't store `isExtender` anywhere and recalculate it on the fly.  Let's rename it
to `pureExtender` or `isPureExtender` since that makes it clear what the difference with
hasExtender is:

```
@property
def pureExtender(self):
    '''
    Read-only boolean property that returns True if an extender is part of the 
    figure but no number is given (a number of 1 means no-number). It is a pure extender.

    >>> n = notation.Figure(1, '#', extender=True)
    >>> n.isExtender
    True
    >>> n.number = 2
    >>> n.isExtender
    False
    '''
    return self.number == 1 and self.hasExtender
```
and then it will always be up-to-date no matter what.

note -- should a pureExtender also require notationString to be '' or None?

Reply:
I see. Not storing this seperatly it makes sense. Thanks for adding this. I will include it.